src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -70,8 +73,30 @@ def create_storage_account(cmd, resource_group_name, account_name, sku=None, loc
 
     if custom_domain:
         params.custom_domain = CustomDomain(name=custom_domain, use_sub_domain=None)
+
+    # Encryption
     if encryption_services:
         params.encryption = Encryption(services=encryption_services)
+
+    if encryption_key_source is not None:
+        params.encryption.key_source = encryption_key_source
+
+    if params.encryption.key_source and params.encryption.key_source == "Microsoft.Keyvault":
+        if params.encryption.key_vault_properties is None:
+            KeyVaultProperties = cmd.get_models('KeyVaultProperties')
+            params.encryption.key_vault_properties = KeyVaultProperties(key_name=encryption_key_name,
+                                                                        key_vault_uri=encryption_key_vault,
+                                                                        key_version=encryption_key_version)
+
+    if identity_type and 'UserAssigned' in identity_type and user_identity_id:
+        params.identity = Identity(type=identity_type, user_assigned_identities={user_identity_id: {}})
+    elif identity_type:
+        params.identity = Identity(type=identity_type)
+    if key_vault_user_identity_id is not None:
```

Comment:
```suggestion
    if key_vault_user_identity_id is not None and params.encryption:
```
To avoid `params.encryption.encryption_identity = ...` crush when `params.encryption` is `None`.
Maybe we can add validator for this.